From 3ee61ee3ca27afdafa41c7787f146a7b3d73dd63 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Sat, 4 Jan 2025 14:53:13 +0200
Subject: [PATCH] ltmain.in: Update info supplied to user for --tag usage

* build-aux/ltmain.in: Make error message more explicit for users,
and add tag name information to help menu.
* tests/infer-tag.at: Add testcase for unsupported compiler use.
---
 build-aux/ltmain.in | 19 +++++++++++++++++--
 tests/infer-tag.at  | 17 +++++++++++++++++
 2 files changed, 34 insertions(+), 2 deletions(-)

diff --git a/build-aux/ltmain.in b/build-aux/ltmain.in
index 8c48f14b..61266d20 100644
--- a/build-aux/ltmain.in
+++ b/build-aux/ltmain.in
@@ -149,6 +149,20 @@ func_help ()
     func_usage_message
     $ECHO "$long_help_message
 
+If a TAG is supplied, it must use one of the tag names below:
+
+    Tag Name        Language Name
+       CC              C
+       CXX             C++
+       GCJ             Java
+       F77             Fortran 77
+       FC              Fortran
+       GO              Go
+       RC              Windows Resource
+
+If you do not see a tag name associated with your programming language, then
+you are using a compiler that $progname does not support.
+
 MODE must be one of the following:
 
        clean           remove files from the build directory
@@ -849,8 +863,9 @@ func_infer_tag ()
 	# was found and let the user know that the "--tag" command
 	# line option must be used.
 	if test -z "$tagname"; then
-	  func_echo "unable to infer tagged configuration"
-	  func_fatal_error "specify a tag with '--tag'"
+	  func_echo "unable to infer tagged configuration with compiler."
+	  func_echo "Possible use of unsupported compiler."
+	  func_fatal_error "specify a tag with '--tag'. For more information, try '$progname --help'."
 #	else
 #	  func_verbose "using $tagname tagged configuration"
 	fi
diff --git a/tests/infer-tag.at b/tests/infer-tag.at
index 079bf7a8..2f4e6777 100644
--- a/tests/infer-tag.at
+++ b/tests/infer-tag.at
@@ -93,3 +93,20 @@ AT_CHECK([$LIBTOOL --mode=compile $GCJ $GCJCFLAGS -c A.java],
 	 [], [ignore], [ignore])
 
 AT_CLEANUP
+
+# Unsupported compiler check
+AT_SETUP([Unsupported compiler inferred tag check])
+AT_KEYWORDS([libtool])
+
+AT_DATA([a.asm],
+[[section .text
+     global _start
+  _start:
+     mov eax,1
+]])
+
+AT_XFAIL_IF([:])
+AT_CHECK([$LIBTOOL --mode=compile nasm -felf64 a.asm],
+	 [0], [ignore], [ignore])
+
+AT_CLEANUP
-- 
2.45.2

